Documentation related to Instance lease feature (#492)
apache/cloudstack-documentation: source/adminguide/virtual_machines.rst

destroyed and recreated, but from the same Template or ISO that was
already in use by the Instance.

Instance Lease
--------------

> Cloudstack provides capability to create instance on lease. Lease denotes a set period for which resource is allocated and upon expiry cleanup is performed.

Reviewer: Explaination of various behaviour scenarios are missing; for example how the feature will work, what happens when the global is disable (i.e. cancels the leases). I think all the various notes on implementation can be put here for the end-user/operator to understand, incld. expected edge-case behaviours.

Author: There are notes included in the docs addressing:
1. Expiry action execution
2. Lease cancellation
3. Delete protection

However, including behaviours as list will provide better clarity. I will update the document.

This feature enables automated cleanup of instances created for specific duration and for specific purpose. This feature gives administrators the ability to automatically reclaim
resources that are no longer needed by expired virtual machines, helping to optimize resource utilization and reduce wastage.

**Configuring lease feature**

The cloud administrator can use global configuration variables to control the behavior of Instance Lease.
To set these variables, API or CloudStack UI can be used:

.. cssclass:: table-striped table-bordered table-hover

======================================= ========================
Configuration                            Description
======================================= ========================
instance.lease.enabled                   Indicates whether to enable the Instance lease featuew, will be applicable only on instances created after lease is enabled. **Default: false**

> instance.lease.scheduler.interval		  Background task interval in seconds that executes Lease expiry action on eligibile expired instances. Default: 3600.

Reviewer: minor nit - additional blank spaces in the start of the description.

instance.lease.alertscheduler.interval	  Background task interval in seconds that executes Lease alert for instances about to be expired in next N days. Default: 86400
instance.lease.alert.daysbefore          Denotes number of days (N) for alert task. Default: 7 days
======================================= ========================

**Using Instance Lease**

> User can associate Lease to an instance during Deployment of instance or modify existing lease by editing instance.

Reviewer: Worth mentioning, the behaviour around what happens:
* When instances are not deployed with leases, can they acquire lease by switching service offering?
* What is allowed for admins & end-users for instances deployed with lease
* Generally when service offering is changed
* When lease is disabled by user or admin for an instance

destroyed and recreated, but from the same Template or ISO that was
already in use by the Instance.

Instance Lease
--------------

Cloudstack provides capability to create instance on lease. Lease denotes a set period for which resource is allocated and upon expiry cleanup is performed.

> This feature enables automated cleanup of instances created for specific duration and for specific purpose. This feature gives administrators the ability to automatically reclaim

Reviewer 2: CloudStack offers the option to create Instances with a Lease. A Lease defines a set time period after which a selected action, such as stopping or destroying the instance, will be automatically performed. This helps optimize cloud resource usage by automatically freeing up resources that are no longer in use. If a user needs an instance only for a limited time, this feature can be useful. When deploying an instance, users can either choose a Compute Offering that includes the Instance Lease feature or enable it specifically for that instance, setting the number of days after which the instance should be stopped or destroyed once their task is complete.

resources that are no longer needed by expired virtual machines, helping to optimize resource utilization and reduce wastage.

> **Configuring lease feature**

Reviewer 2: **Configuring Instance Lease feature**

The cloud administrator can use global configuration variables to control the behavior of Instance Lease.
To set these variables, API or CloudStack UI can be used:

======================================= ========================
Configuration                            Description
======================================= ========================

> instance.lease.enabled                   Indicates whether to enable the Instance lease featuew, will be applicable only on instances created after lease is enabled. **Default: false**

Reviewer 2: featuew -> feature

destroyed and recreated, but from the same Template or ISO that was
already in use by the Instance.

Instance Lease
--------------

CloudStack offers the option to create Instances with a Lease. A Lease defines a set time period after which a selected action, such as stopping or destroying the instance, 
will be automatically performed. This helps optimize cloud resource usage by automatically freeing up resources that are no longer in use. 

> If a user needs an instance only for a limited time, this feature can be useful. When deploying an instance, users can either choose a Compute Offering that includes 

Reviewer 3: ```suggestion
It is better to lease an instance if you need it only for a limited time. When deploying an instance, you can either choose a Compute Offering that includes 
```

However, it is not possible to enable the lease on an instance after it has already been deployed.

From UI:

.. image:: /_static/images/edit_instance_lease.png
   :width: 400px
   :align: center
   :alt: Edit Instance lease dialog

Using API:

.. code:: bash

   cmk update virtualmachine id=fa970d19-8340-455c-a9fb-569205954fdc leaseduration=20 leaseexpiryaction=DESTROY

To disable lease using API:

.. code:: bash

   cmk update virtualmachine id=fa970d19-8340-455c-a9fb-569205954fdc leaseduration=-1

.. note:: DESTORY action will ignore instance if deleteprotection is enabled for it.

.. note:: When the feature is disabled, the lease associated with instances is cancelled. Re-enabling the feature will not automatically reapply the lease to previously grandfathered instances.

**Instance Lease Events**

Lease feature generates various events to help in auditing and monitoring:

=================== ========================
Event Type           Description
=================== ========================
VM.LEASE.EXPIRED	   Event is generated at lease expiry
VM.LEASE.DISABLED	   Denotes if lease is disabled by user/admin

> VM.LEASE.CANCELLED   When lease is cancelled (feature gets disabled)

Reviewer 3: check indentation
